aten/src/ATen/native/native_functions.yaml

```diff
@@ -5674,12 +5674,12 @@
 - func: choose_qparams_optimized(Tensor input, int numel, int n_bins, float ratio, int bit_width) -> (Tensor, Tensor)
   variants: function
 
-- func: _autocast_to_reduced_precision(Tensor(a) self, bool cuda_enabled, bool cpu_enabled, ScalarType cuda_dtype, ScalarType cpu_dtype) -> Tensor(a)
-  variants: method
+- func: _autocast_to_reduced_precision(Tensor? self, bool cuda_enabled, bool cpu_enabled, ScalarType cuda_dtype, ScalarType cpu_dtype) -> Tensor
+  variants: function 
```

Comment:
My question is: do we want to have an overload for autocast ops, so we are properly having two schema for each casting op, like
```
_autocast_to_reduced_precision.optional(Tensor? self, ...) -> Tensor?
_autocast_to_reduced_precision(Tensor self, ...) -> Tensor
```

I'm currently faking this by manually setting output value type to be optional (e.g. here: https://github.com/pytorch/pytorch/pull/75094/files#diff-497ea45608a44bf0d1a3046c2627f9901ad8b94e2645db8916cd0969a0e9c0f9R183-R185)
This is necessary, otherwise, consumer of casted optional tensor can't match its function signature.

e.g. vvv in the example David posted in #75002, autocast op changes bias dtype to Tensor instead of Optional tensor, so later alias analysis can't find a conv2d op that takes bias as a tensor, instead of a optional tensor
```
[DUMP profiling_graph_executor_impl.cpp:669] Profiled Graph:                                                               
[DUMP profiling_graph_executor_impl.cpp:669] graph(%inp.1 : Tensor,                                                        
[DUMP profiling_graph_executor_impl.cpp:669]       %weight.1 : Tensor,                                                     
[DUMP profiling_graph_executor_impl.cpp:669]       %bias.1 : Tensor?):                                                     
[DUMP profiling_graph_executor_impl.cpp:669]   %3 : int[] = prim::Constant[value=[3, 3]]()                                 
[DUMP profiling_graph_executor_impl.cpp:669]   %4 : int[] = prim::Constant[value=[1, 1]]()                                 
[DUMP profiling_graph_executor_impl.cpp:669]   %5 : int = prim::Constant[value=5]() # david_repro.py:6:53                  
[DUMP profiling_graph_executor_impl.cpp:669]   %6 : bool = prim::Constant[value=1]() # david_repro.py:6:41                 
[DUMP profiling_graph_executor_impl.cpp:669]   %7 : int = prim::Constant[value=1]() # david_repro.py:7:61                  
[DUMP profiling_graph_executor_impl.cpp:669]   %8 : str = prim::Constant[value="cuda"]() # /raid/pytorch_upstream/torch/cuda/amp/autocast_mode.py:22:26
[DUMP profiling_graph_executor_impl.cpp:669]   %9 : bool = prim::Constant[value=0]()                                       
[DUMP profiling_graph_executor_impl.cpp:669]   %10 : int = prim::Constant[value=15]()                                      
[DUMP profiling_graph_executor_impl.cpp:669]   %11 : __torch__.torch.cuda.amp.autocast_mode.autocast = prim::CreateObject()
[DUMP profiling_graph_executor_impl.cpp:669]    = prim::SetAttr[name="_enabled"](%11, %6)                                  
[DUMP profiling_graph_executor_impl.cpp:669]    = prim::SetAttr[name="device"](%11, %8)                                    
[DUMP profiling_graph_executor_impl.cpp:669]    = prim::SetAttr[name="fast_dtype"](%11, %5)                                
[DUMP profiling_graph_executor_impl.cpp:669]   %12 : __torch__.torch.cuda.amp.autocast_mode.autocast = prim::Enter(%11) 
[DUMP profiling_graph_executor_impl.cpp:669]   %18 : Tensor = prim::profile[profiled_type=Tensor, seen_none=0](%inp.1)  
[DUMP profiling_graph_executor_impl.cpp:669]   %13 : Tensor = aten::_autocast_to_reduced_precision(%18, %6, %9, %5, %10)
[DUMP profiling_graph_executor_impl.cpp:669]   %19 : Tensor = prim::profile[profiled_type=Tensor, seen_none=0](%weight.1)
[DUMP profiling_graph_executor_impl.cpp:669]   %14 : Tensor = aten::_autocast_to_reduced_precision(%19, %6, %9, %5, %10)
[DUMP profiling_graph_executor_impl.cpp:669]   %20 : Tensor? = prim::profile[profiled_type=Tensor, seen_none=0](%bias.1)
[DUMP profiling_graph_executor_impl.cpp:669]   %15 : Tensor = aten::_autocast_to_reduced_precision(%20, %6, %9, %5, %10)
[DUMP profiling_graph_executor_impl.cpp:669]   %21 : Tensor = prim::profile[profiled_type=Tensor, seen_none=0](%13)        
[DUMP profiling_graph_executor_impl.cpp:669]   %22 : Tensor = prim::profile[profiled_type=Tensor, seen_none=0](%14)        
[DUMP profiling_graph_executor_impl.cpp:669]   %23 : Tensor = prim::profile[profiled_type=Tensor, seen_none=0](%15)        
[DUMP profiling_graph_executor_impl.cpp:669]   %16 : Tensor = aten::conv2d(%21, %22, %23, %4, %3, %4, %7) # david_repro.py:7:15
[DUMP profiling_graph_executor_impl.cpp:669]   %17 : Tensor = prim::Exit(%11)                                              
[DUMP profiling_graph_executor_impl.cpp:669]   %24 : Tensor = prim::profile[profiled_type=Tensor, seen_none=0](%16)        
[DUMP profiling_graph_executor_impl.cpp:669]    = prim::profile()                                                          
[DUMP profiling_graph_executor_impl.cpp:669]   return (%24)                                                                
[DUMP profiling_graph_executor_impl.cpp:669] Profiled Graph:                                                               
[DUMP profiling_graph_executor_impl.cpp:669] graph(%self.1 : __torch__.torch.cuda.amp.autocast_mode.autocast):             
[DUMP profiling_graph_executor_impl.cpp:669]    = prim::profile()                                                          
[DUMP profiling_graph_executor_impl.cpp:669]   return (%self.1)                                                            
[DUMP profiling_graph_executor_impl.cpp:669] Profiled Graph:                                                               
[DUMP profiling_graph_executor_impl.cpp:669] graph(%self : __torch__.torch.cuda.amp.autocast_mode.autocast,                
[DUMP profiling_graph_executor_impl.cpp:669]       %exc_type : Any,                                                        
[DUMP profiling_graph_executor_impl.cpp:669]       %exc_val : Any,                                                         
[DUMP profiling_graph_executor_impl.cpp:669]       %exc_tb : Any):                                                         
[DUMP profiling_graph_executor_impl.cpp:669]   %4 : NoneType = prim::Constant() # /raid/pytorch_upstream/torch/cuda/amp/autocast_mode.py:35:12
[DUMP profiling_graph_executor_impl.cpp:669]    = prim::profile()                                                          
[DUMP profiling_graph_executor_impl.cpp:669]   return (%4)                                                                 
bias is none                                                                                                               
eager                                                                                                                      
jit                                                                                                                        
bias is not none                                                                                                           
eager                                                                                                                      
jit                                                                                                                        
Traceback (most recent call last):                                                                                         
  File "david_repro.py", line 25, in <module>                                                                              
    fn_s(inp, weight, bias)                                                                                                
RuntimeError: 0 INTERNAL ASSERT FAILED at "/raid/pytorch_upstream/torch/csrc/jit/ir/alias_analysis.cpp":613, please report a bug to PyTorch. We don't have an op for aten::conv2d but it isn't a special case.  Argument types: Tensor, Tensor, Tensor, int[], int[], int[], int,
```